modules/sources/contracts/fetch-tokenpair-data.ts

```diff
@@ -98,12 +102,27 @@ export async function fetchTokenPairData(
             addAToBPriceCallsToMulticaller(tokenPair, routerAddress, multicallerRouter);
         }
     });
+    
+    let bptTokenPairs = generateBptTokenPairs(pools);
+    bptTokenPairs.forEach((bptTokenPair) => {
+        if (bptTokenPair.valid) {
+            // prepare swap amounts in
+            // tokenA->tokenB with 1% of tokenA balance
+            bptTokenPair.aToBAmountIn = parseUnits(bptTokenPair.tokenA.balance, bptTokenPair.tokenA.decimals) / 100n;
+            // tokenA->tokenB with 100USD worth of tokenA
+            const oneHundredUsdOfTokenA = (parseFloat(bptTokenPair.tokenA.balance) / bptTokenPair.tokenA.balanceUsd) * 100;
+            bptTokenPair.effectivePriceAmountIn = parseUnits(`${oneHundredUsdOfTokenA}`, bptTokenPair.tokenA.decimals);
```

Comment:
One thing that called my attention is that you're using BPT as the tokenA, which is the one that you need to know the `balanceUsd`. For some reason I think this info is not as available/updated for BPTs compared to "regular" pool tokens.
We could reverse the order and have BPT being tokenB for this whole flow without much effort, right @lgahdl?
@franzns - do you see benefit on this change? Or is it ok to leave as is?

Reply:
Never mind. I saw that you're using totalLiquidity and I think this should be ok 👍